toolkit/tools/scheduler/schedulerutils/preparerequest.go

```diff
@@ -76,6 +93,17 @@ func buildNodesToRequests(pkgGraph *pkggraph.PkgGraph, buildState *GraphBuildSta
 		}
 
 		defaultNode := buildNodes[0]
+
+		// Check if we already queued up this build node for building.
+		if buildState.IsSRPMBuildActive(defaultNode.SRPMFileName()) || buildState.IsNodeProcessed(defaultNode) {
+			err = fmt.Errorf("unexpected duplicate build for (%s)", defaultNode.SRPMFileName())
+			// Temporarily ignore the error, this state is unexpected but not fatal. Error return will be
+			// restored later once the underlying cause of this error is fixed.
+			logger.Log.Errorf(err.Error())
+			continue
+			//return
```

Comment:
Nit-pick: considering the robust and clear comment above, I think this is not needed - we'll know how to update the code, once we find the bug. And who knows, maybe we'll panic since this is something that should never occur, or do something else.
```suggestion
```